docs/Running-Mastodon/Production-guide.md

```diff
@@ -239,6 +292,62 @@ You may want to run `which bundle` first and copypaste that full path instead of
 
 You can edit the cronjob file for the `mastodon` user by running `sudo crontab -e -u mastodon` (outside of the mastodon user).
 
+## Monitoring (Optional)
+You can monitor your server's status using [netdata](https://github.com/firehol/netdata/wiki/Installation)! Installation looks different depending on your OS - find details at their website, or follow along here if you're running Ubuntu 16.04:
+
+    sudo apt-get install zlib1g-dev uuid-dev libmnl-dev gcc make git autoconf autoconf-archive autogen automake pkg-config curl
+    git clone https://github.com/firehol/netdata.git --depth=1
+    cd netdata
+    ./netdata-installer.sh
+
+Netdata should now be running on localhost:19999. To expose it via nginx we need to add a few rules to our nginx configuration.
+
+1. Inside of the `http` directive, as a sibling to the various `server` directives we set up above, add this:
+    ```
+    upstream netdata {
+        server 127.0.0.1:19999;
+        keepalive 64;
+    }
+    ```
+
+2. Inside of the `server` directive that's handling https, next to all of the other locations, add the following two:
+    ```
+    location /netdata {
+      return 301 /netdata/;
+    }
+
+    location ~ /netdata/(?<ndpath>.*) {
+      proxy_redirect off;
+      proxy_set_header Host $host;
+
+      proxy_set_header X-Forwarded-Host $host;
+      proxy_set_header X-Forwarded-Server $host;
+      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+      proxy_http_version 1.1;
+      proxy_pass_request_headers on;
+      proxy_set_header Connection "keep-alive";
+      proxy_store off;
+      proxy_pass http://netdata/$ndpath$is_args$args;
+
+      gzip on;
+      gzip_proxied any;
+      gzip_types *;
+    }
+    ```
+
+Then `sudo systemctl restart nginx` and navigate to https://YOUR_DOMAIN/netdata to see realtime monitoring data.
+
+## Updating Your SSL Certificate
+Remember above when we set up our LetsEncrypt certificate using a standalone server? In three months, when it tries to auto-renew that certificate, it's going to fail because we now have something else listening on those ports. (namely, nginx and our mastodon app). 
+
+So technically your certificate is fine - it will work for three months. But if you want it to auto-renew (and, like, you do) then once everything is working go ahead and re-run `certbot certonly`. This time, choose option (1) (Place files in webroot directory (webroot)). It'll tell you that your cert isn't close to expiration, and ask you if you're sure - say yes. Enter a new webroot (press `1` again when prompted) and when it asks for the path, give it `/home/users/mastodon/live/public` (or whatever your equivalent path is). This will allow the whole thing to re-run and verify your mastodon instance.
```

Comment:
Turns out I forgot something else. You'll need to reload your `nginx` after renewing; otherwise the new certificate won't be used. To do that, you need to change the cronjob from:

    0 */12 * * * root test -x /usr/bin/certbot -a \! -d /run/systemd/system && perl -e 'sleep int(rand(3600))' && certbot -q renew

to:

    0 */12 * * * root test -x /usr/bin/certbot -a \! -d /run/systemd/system && perl -e 'sleep int(rand(3600))' && certbot -q renew --renew-hook "service nginx reload"

That's the default cronjob on Debian (from `/etc/cron.d/certbot`). Not 100% sure if it's the same on Ubuntu.